codecov_auth/authentication/repo_auth.py

```diff
@@ -243,87 +244,131 @@ def authenticate_credentials(self, token):
         )
 
 
-class TokenlessAuthentication(authentication.TokenAuthentication):
-    """This is named ironically, but provides authentication for tokenless uploads.
-    Tokenless only works in FORKs of PUBLIC repos.
-    It allows PRs from external contributors (forks) to upload coverage
-    for the upstream repo when running in a PR.
+tokenless_auth_failed_message = "Not valid tokenless upload"
 
-    While it uses the same "shell" (authentication.TokenAuthentication)
-    it doesn't really rely on tokens to authenticate.
-    """
 
-    auth_failed_message = "Not valid tokenless upload"
-    rate_limit_failed_message = "Tokenless has reached GitHub rate limit. Please upload using a token: https://docs.codecov.com/docs/adding-the-codecov-token."
+def check_tokenless_header(request, branch=None):
+    tokenless = request.headers.get("X-Tokenless", "")
+
+    if not tokenless:
+        return False
+
+    if ":" not in tokenless:
+        raise exceptions.AuthenticationFailed(tokenless_auth_failed_message)
+
+    # make sure it's backwards compatible with the old way that
```

Comment:
It can't be compatible if validation fails (in the line above) because "the old way" doesn't include a `:` in the `X-Tokenless` header, ...right?